dotnet/Services/CybersourcePaymentService.cs

```diff
@@ -398,6 +430,96 @@ public async Task<CreatePaymentResponse> CreatePayment(CreatePaymentRequest crea
+                            PickupStoreInfo = new PickupStoreInfo
+                            {
+                                AdditionalInfo = selectedSla.PickupStoreInfo.AdditionalInfo,
+                                Address = selectedSla.PickupStoreInfo.Address != null ? new VtexAddress
+                                {
+                                    AddressId = selectedSla.PickupStoreInfo.Address.AddressId,
+                                    AddressType = selectedSla.PickupStoreInfo.Address.AddressType,
+                                    City = selectedSla.PickupStoreInfo.Address.City,
+                                    Complement = selectedSla.PickupStoreInfo.Address.Complement,
+                                    Country = selectedSla.PickupStoreInfo.Address.Country,
+                                    GeoCoordinates = selectedSla.PickupStoreInfo.Address.GeoCoordinates,
+                                    IsDisposable = selectedSla.PickupStoreInfo.Address.IsDisposable,
+                                    Neighborhood = selectedSla.PickupStoreInfo.Address.Neighborhood,
+                                    Number = selectedSla.PickupStoreInfo.Address.Number,
+                                    PostalCode = selectedSla.PickupStoreInfo.Address.PostalCode,
+                                    ReceiverName = selectedSla.PickupStoreInfo.Address.ReceiverName,
+                                    Reference = selectedSla.PickupStoreInfo.Address.Reference,
+                                    State = selectedSla.PickupStoreInfo.Address.State,
+                                    Street = selectedSla.PickupStoreInfo.Address.Street
+                                } : null,
+                                DockId = selectedSla.PickupStoreInfo.DockId,
+                                FriendlyName = selectedSla.PickupStoreInfo.FriendlyName,
+                                IsPickupStore = selectedSla.PickupStoreInfo.IsPickupStore
+                            },
+                            PolygonName = selectedSla.PolygonName,
+                            ShippingEstimate = selectedSla.ShippingEstimate,
+                            ShippingEstimateDate = selectedSla.ShippingEstimateDate,
+                            Price = selectedSla.Price,
+                            Tax = selectedSla.Tax,
+                            TransitTime = selectedSla.TransitTime
+                        };
+
+                        requestWrapper.Shipping.DeliveryIds.AddRange(selectedSla.DeliveryIds);
+                        requestWrapper.Shipping.CourierName = string.Join(", ", selectedSla.DeliveryIds.Select(d => d.CourierName).Distinct().ToArray());
+                    }
+
+                    if (requestWrapper.Totals != null)
+                    {
+                        requestWrapper.Totals.Discounts += (decimal)(vtexOrder.Totals.Where(i => i.Id.Equals("Discounts")).Sum(t => t.Value)) / 100;
```

Comment:
For performance, I think we can iterate through the element `i` only once and run the 4 operations as they all operate on the same object

Reply:
Changed